packages/web/src/features/users/AcceptInviteView.tsx

```diff
@@ -0,0 +1,112 @@
+import { useTranslation } from 'react-i18next';
+import { useForm } from 'react-hook-form';
+import Card from '../../shared/components/Card';
+import FormLabel from '../../shared/components/form/FormLabel';
+import TextInput from '../../shared/components/form/TextInput';
+import View from '../../shared/components/View';
+import ViewTitle from '../../shared/components/ViewTitle';
+import apiClient from '../../shared/apiClient';
+import Form, { SubmitHandler } from '../../shared/components/form/Form';
+import InputError from '../../shared/components/form/InputError';
+import Button from '../../shared/components/actions/Button';
+import SubmittingIndicator from '../../shared/components/form/SubmittingIndicator';
+import { useFlash } from '../../shared/hooks/flash';
+import useAuth from '../../shared/hooks/useAuth';
+import {
+  LoaderFunctionArgs,
+  useLoaderData,
+  useNavigate,
+  useSearchParams,
+} from 'react-router-dom';
+import { QueryClient } from '@tanstack/query-core';
+import { useQuery } from '@tanstack/react-query';
+
+const createInviteQuery = (token?: string) => ({
+  queryKey: ['invite', token],
+  queryFn: () => apiClient.auth.getInvite(token ?? ''),
+});
+
+export const acceptInviteLoader =
+  (queryClient: QueryClient) =>
+  ({ request }: LoaderFunctionArgs) => {
+    const url = new URL(request.url);
+    const token = url.searchParams.get('token') ?? undefined;
+    return queryClient.ensureQueryData(createInviteQuery(token));
+  };
+
+const useInviteQuery = (token?: string) => {
+  const initialData = useLoaderData() as Awaited<
+    ReturnType<ReturnType<typeof acceptInviteLoader>>
+  >;
+  return useQuery({
+    ...createInviteQuery(token),
+    initialData,
+  });
+};
```

Comment:
This pattern of using both react-query and loaders from react-router I got from [this blog post](https://tkdodo.eu/blog/react-query-meets-react-router). I think we should use this as our primary data loading pattern. I'm going to document and streamline this in another PR.

Basically, the loader primes the cache and allows the view to wait to load until that request completes. Then after the view loads, the query will have data immediately because it is in the cache, but can be updated when it becomes stale.